src/leveragers/Leverager.sol

```diff
@@ -0,0 +1,418 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.19;
+
+import "../interfaces/ILeverager.sol";
+import "lib/openzeppelin-contracts/contracts/access/Ownable.sol";
+import "lib/openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+import "../interfaces/wETH/IWETH.sol";
+import "../interfaces/alchemist/IAlchemistV2.sol";
+import "../interfaces/alchemist/ITokenAdapter.sol";
+import "../interfaces/uniswap/TransferHelper.sol";
+import "../interfaces/curve/ICurveFactory.sol";
+
+//import console log
+import "forge-std/console.sol";
+
+abstract contract Leverager is ILeverager, Ownable {
+    address public yieldToken;
+    address public underlyingToken;
+    address public debtToken;
+    address public flashLoan;
+    address public debtSource;
+    address public dex;
+    IAlchemistV2 alchemist;
+    
+    address constant curveEth = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
```

Comment:
If this is an abstraction from the underlying exchange, should any Curve specific logic exist in here? Same with wETH logic.

Reply:
You're correct that I left everything that was common to both leveragers in the abstraction, without it we'd need another level of abstraction as I don't think that the weth and curve logic should be copy pasted in both leverager implementations. Using the current (probably imperfect based on your comments) level of abstraction reduces by a lot the amount of similar code between the implementations.
As to why it's made in such a way, is because I don't have a handle good enough on the project to know if the curve functionality is common to all implementations or not, so I processed this using a diff between both leveragers and substracted the common part to build the abstraction.
From here we have three options, either this PR serves a different purpose than expected, removing redundant code, or we extend the current version with a higher level of abstraction without the curve logic so implementers can pick the one closer to what they need, or we can remove the curve logic from the abstraction and copy paste the same code back into both of the current implementations.